bob/src/cluster/quorum.rs

```diff
@@ -77,6 +78,38 @@ impl Quorum {
         }
     }
 
+    async fn delete_on_nodes(&self, key: BobKey, with_aliens: bool) -> Result<(), Error> {
+        debug!("DELETE[{}] ~~~DELETE LOCAL NODE FIRST~~~", key);
+        let mut local_put_ok = 0_usize;
+        let mut failed_nodes = Vec::new();
+        let res = delete_at_local_node(&self.backend, key, with_aliens).await;
+        if let Err(e) = res {
+            error!("{}", e);
+            failed_nodes.push(self.mapper.local_node_name().to_owned());
+        } else {
+            local_put_ok += 1;
+            debug!("DELETE[{}] local node delete successful", key);
+        }
+
+        debug!("DELETE[{}] ~~~DELETE TO REMOTE NODES~~~", key);
+        let errors = self.delete_at_remote_nodes(key, with_aliens).await;
+        let all_count = self.mapper.get_target_nodes_for_key(key).len();
```

Comment:
Total number of nodes depends on `with_aliens` flag. I suggest to return number of affected nodes from `delete_at_remote_nodes` inside tuple. In this case `all_count` can be calculated as `affected_nodes + if delete_on_local_node {1} else {0}`